ts/src/sourcemapper/sourcemapper.ts

```diff
@@ -185,56 +173,68 @@ export class SourceMapper {
    *   and line information.
    *
    *   If the given input file does not have mapping information associated
-   *   with it then null is returned.
+   *   with it then the input location is returned.
    */
-  mappingInfo(inputPath: string, lineNumber: number, colNumber: number):
-      MapInfoOutput|null {
-    inputPath = path.normalize(inputPath);
+  mappingInfo(location: GeneratedLocation): SourceLocation {
+    const inputPath = path.normalize(location.file);
     const entry = this.getMappingInfo(inputPath);
     if (entry === null) {
-      return null;
+      return location;
     }
 
-    const sourcePos = {
-      source: path.relative(path.dirname(entry.mapFile), inputPath)
-                  .replace(/\\/g, '/'),
-      line: lineNumber + 1,  // the SourceMapConsumer expects the line number
-                             // to be one-based but expects the column number
-      column: colNumber      // to be zero-based
-    };
+    const generatedPos = {line: location.line, column: location.column};
 
     // TODO: Determine how to remove the explicit cast here.
     const consumer: sourceMap.SourceMapConsumer =
         entry.mapConsumer as {} as sourceMap.SourceMapConsumer;
-    const allPos = consumer.allGeneratedPositionsFor(sourcePos);
-
-    const mappedPos: sourceMap.LineRange =
-        consumer.generatedPositionFor(sourcePos);
 
+    const pos = consumer.originalPositionFor(generatedPos);
+    if (pos.source === null) {
+      return location;
+    }
     return {
-      file: entry.outputFile,
-      line: mappedPos.line - 1,  // convert the one-based line numbers returned
-                                 // by the SourceMapConsumer to the expected
-                                 // zero-based output.
-      column: mappedPos.column   // SourceMapConsumer uses
-                                 // zero-based column
-                                 // numbers which is the
-                                 // same as the expected
-                                 // output
+      file: path.resolve(entry.mapFileDir, pos.source),
+      line: pos.line || undefined,
+      name: pos.name || location.name,
+      column: pos.column,
     };
   }
 }
 
-export async function create(sourcemapPaths: string[]): Promise<SourceMapper> {
+export async function create(sourceMapSearchPath: string[]):
```

Comment:
Nit: consider dropping "sourceMap" prefix from names where it's clear from the context. E.g. sourceMapSearchPath can be shortened to searchPath without affecting readability I think, similar for some local vars below.

Reply:
I left the "map" prefix in some places (since this file imports "path", I used "mapPath" in places; and also used "mapFiles" and "mapPath" for consistency).